barcodes_generator_abstract/models/barcode_rule.py

```diff
@@ -49,6 +50,14 @@ class BarcodeRule(models.Model):
         "a new record in the mixed model.",
     )
 
+    is_default_rule = fields.Boolean("Is Default Barcode Rule", copy=False)
+
+    @api.constrains("is_default_rule")
+    def _check_is_default_rule(self):
+        default_rule = self.search_count([("is_default_rule", "=", True)])
+        if default_rule > 1:
+            raise UserError(_("Only one rule can be set as default."))
```

Comment:
This check looks wrong. 
We can have a default rule for partner model, and another one for product model.